src/Inventory/Asset/Printer.php

```diff
@@ -365,8 +365,8 @@ public function handleMetrics()
      */
     public static function needToBeUpdatedFromDiscovery(CommonDBTM $item, $val)
     {
-        if (property_exists($val, 'ips') && isset($val->ips[0])) {
-            $ip = $val->ips[0];
+        if (property_exists($val, 'ip')) {
```

Comment:
Are you sure there is always a `$val->ip` if there is a `$val->ips[0]`? I'm really unsure

Reply:
I will wait for @periodistacg feedback on these tests.
If so, I already have another solution 
Filter all ```ips``` with ```Blacklist```

Because for now this method only check first entry of ```ips```

```xml
<IPS>
  <IP>0.0.0.0</IP>  <-- this is use for check if printer need to be updated
  <IP>10.100.51.207</IP>
  <IP>127.0.0.1</IP>
</IPS>
```
and ```Inventory``` understands that he must update the ```network``` part